src/State/Processor/RespondProcessor.php

```diff
@@ -150,4 +165,21 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             $headers
         );
     }
+
+    private function getAllowedMethods(?string $resourceClass, ?string $currentUriTemplate): array
+    {
+        $allowedMethods = self::DEFAULT_ALLOWED_METHOD;
+        if (null !== $currentUriTemplate && null !== $resourceClass && $this->resourceClassResolver->isResourceClass($resourceClass)) {
+            $resourceMetadataCollection = $this->resourceCollectionMetadataFactory ? $this->resourceCollectionMetadataFactory->create($resourceClass) : new ResourceMetadataCollection($resourceClass);
```

Comment:
```suggestion
            $resourceMetadataCollection = $this->resourceCollectionMetadataFactory ? $this->resourceCollectionMetadataFactory->create($resourceClass) : new ResourceMetadataCollection($resourceClass);
```

Better not set the extra headers at all if the factory is missing.

It's an optional feature after all.